### DISCUSSION TITLE
how to set callbackUrl and  callbackBody

### DISCUSSION THREAD

**Discussion by dengao123** [minio/minio#17460] [Q&A] [answered]
When I upload a file to aliyun, an upload callback is triggered. However, when I upload to minio, the upload callback cannot be triggered. How can I set it up?

**Comment by klauspost** [comment, marked as answer]
You can set up either [audit logs](https://min.io/docs/minio/kubernetes/upstream/operations/monitoring/minio-logging.html#minio-logging-publish-audit-logs) or [bucket notifications](https://min.io/docs/minio/kubernetes/upstream/administration/monitoring.html#bucket-notifications).

The S3 API does not itself provide callbacks.
